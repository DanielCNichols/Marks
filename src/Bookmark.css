* {
  box-sizing: border-box;
  text-decoration: none;
  list-style: none;
}

.tooltiptext {
  display: none;
}

.bookmark-item {
  width: 100%;
  display: grid;
  border: 2px solid white;
  grid-template-columns: 43% 43% 14%;
  grid-template-rows: 1fr 1fr 1fr;
  background-color: #252525;
  height: auto;
  margin: 10px auto;
  padding: 0;
}

.head {
  grid-column-start: span 2;
  grid-row-start: 1;
  font-size: 1.3em;
  display: flex;
  justify-content: center;
  align-items: center;
  word-wrap: break-word;
}

.head h3 {
  margin: 0;
  padding: 5px;
}

.rating {
  grid-column-start: span 2;
  grid-row-start: 2;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.1em;
  color: rgba(255, 255, 255, 0.8);
}

.item-controls-container {
  grid-column-start: 3;
  grid-column-end: 4;
  grid-row-start: span 3;
  margin-left: auto;
  width: 100%;
}

.item-controls {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  background-color: #404140;
  width: auto;
  margin: 0 auto;
}

.item-controls svg {
  margin: 10px 0;
  color: rgba(255, 255, 255, 0.8);
  font-size: 2em;
  padding: 5px;
}

.item-expand {
  grid-row-start: 3;
  grid-column-start: 1;
  grid-column-end: 3;
  text-align: left;
  width: 100%;
  padding-left: 5px;
  display: flex;
  align-items: center;
  border-top: 1px solid rgba(255, 255, 255, 0.3);
}

.item-expand svg {
  margin: 0 10px;
  font-size: 1.3em;
}

.bookmark-item-expanded {
  width: 100%;
  display: grid;
  border: 2px solid white;
  grid-template-columns: 1fr, 1fr;
  grid-template-rows: 1fr 1fr 1fr 20%;
  background-color: #252525;
  height: auto;
  margin: 10px auto;
}

.head-expanded {
  grid-row-start: 1;
  word-wrap: break-word;
  text-align: left;
  display: flex;
  align-items: center;
  padding: 0 20px;
  border-bottom: 2px solid rgba(255, 255, 255, 0.5);
}

.head-expanded h3 {
  margin: 0;
}

.less {
  color: #23a840;
}

.rating-expanded {
  grid-row-start: 1;
  grid-column-start: 2;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding: 0 20px;
  border-bottom: 2px solid rgba(255, 255, 255, 0.5);
}

.desc {
  grid-row-start: 2;
  grid-column-start: span 2;
  padding: 0 10px;
  word-wrap: break-word;
}

.url {
  grid-row-start: 3;
  grid-column-start: span 2;
  padding: 0 10px;
  word-wrap: break-word;
  width: 100%;
  margin: 0 auto;
}

.item-controls-container-expand {
  grid-row-start: 4;
  grid-column-start: span 2;
  border-top: 1px solid white;
  display: flex;
  align-items: center;
  background-color: #404140;
  width: 100%;
}

.item-controls-container-expand svg {
  font-size: 1.3em;
}

.item-controls-expand {
  width: 100%;
  display: flex;
  color: rgba(255, 255, 255, 0.8);
  justify-content: space-around;
}

.less {
  display: flex;
  align-items: center;
}

.bookmark-form {
  border: 2px solid white;
  display: grid;
  grid-template-rows: 1fr repeat(5);
  grid-template-columns: 1fr 1fr;
  background-color: #252525;
  padding: 0 10px;
  margin: 10px 0;
}

.bookmark-form select {
  width: 100%;
  padding: 5px;
  height: 30px;
  margin: 5px;
}

.bookmark-form label {
  display: flex;
  flex-direction: column;
  margin: 10px 0;
  color: #68cffa;
  text-align: left;
}

.edit-form-url {
  grid-row-start: 2;
  grid-column-start: span 2;
}

.edit-form-desc {
  grid-row-start: 3;
  grid-column-start: span 2;
}

.edit-form-controls {
  grid-row-start: 4;
  grid-column-start: span 2;
}

input,
textarea {
  margin: 5px 0;
  padding: 5px;
  border-radius: 4px;
}

.edit-form-controls {
  display: flex;
  justify-content: space-around;
  margin-bottom: 10px;
  font-size: 1.5em;
}

.edit-form-controls button {
  background: transparent;
  border: none;
  font-size: 1.2em;
}

.edit-form-controls button > span {
  color: rgba(255, 255, 255, 0.8);
  font-size: 1rem;
  display: block;
  width: auto;
  margin: 0 auto;
}

.save {
  color: #23a840;
}

.cancel {
  color: red;
}

@media screen and (min-width: 800px) {
  .bookmark-item {
    width: 85%;
    min-width: 355px;
    grid-template-columns: 30% 70%;
    grid-template-rows: 1fr 1fr 1fr;
  }

  .bookmark-item-expanded {
    width: 85%;
    min-width: 355px;
  }

  .item-controls-container {
    grid-column-start: 2;
    grid-row-start: 3;
    width: 100%;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    background-color: #404140;
  }

  .item-controls {
    flex-direction: row;
    align-items: center;
    justify-content: space-around;
    background-color: #404140;
    border-top: 1px solid rgba(255, 255, 255, 0.3);
    height: 100%;
  }

  .item-controls svg {
    padding: 0;
    margin: 0;
    font-size: 1.2em;
  }

  .tooltip {
    position: relative;
    width: auto;
    height: 50%;
  }

  .tooltiptext {
    visibility: hidden;
    width: auto;
    display: flex;
    background-color: black;
    color: rgba(255, 255, 255, 0.8);
    text-align: center;
    border-radius: 6px;
    padding: 5px 0;
    position: absolute;
    left: -70%;
    opacity: 0;
    transition: 0.5s;
    transition-delay: 0.5s;
  }

  .tooltip:hover .tooltiptext {
    visibility: visible;
    position: absolute;
    border: 1px solid rgba(255, 255, 255, 0.5);
    padding: 3px 5px;
    opacity: 1;
  }

  .item-controls div:nth-child(2) > svg:hover {
    color: red;
  }

  .item-controls-expand div:nth-child(3) > svg:hover {
    color: red;
  }

  .item-controls div:nth-child(3) > a:hover svg {
    color: #68cffa;
  }

  .item-controls-expand div:nth-child(4) > a:hover svg {
    color: #68cffa;
  }

  .item-controls svg:nth-child(1):hover {
    color: #23a841;
  }

  .item-controls-expand div:nth-child(2) > svg:hover {
    color: #23a841;
  }

  .item-expand {
    grid-column-start: 1;
    grid-column-end: 2;
    background-color: #404140;
  }

  .item-expand svg {
    margin: 0;
  }
}
