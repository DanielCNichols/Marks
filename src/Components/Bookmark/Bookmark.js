import React, { useState } from 'react';
import { MdDelete, MdEdit, MdExpandMore, MdExpandLess } from 'react-icons/md';
import { TiArrowForward } from 'react-icons/ti';
import ApiService from '../../Services/api-service';
import RatingSpan from '../RatingSpan/RatingSpan';
import s from './Bookmark.module.css';
import Modal from '../Modal/Modal';
import EditForm from '../EditForm/EditForm';

/* //! Lets get weird. Let's make the actual bookmark editable in place.  WE can use role= textbox and contentEditable along with custom javascript to handle this. Don't forget to render the errors!*/

export default function Bookmark({
  bookmark,
  removeBookmark,
  updateBookmark,
  editToggle,
}) {
  const [expanded, setExpanded] = useState(false);
  const [editing, setEditing] = useState(false);
  const [error, setError] = useState(null);
  const [inputs, setInputs] = useState({
    title: bookmark.title,
    rating: bookmark.rating,
    url: bookmark.url,
    desc: bookmark.desc,
  });

  const handleCancel = () => {
    setEditing(false);
  };

  const handleExpand = () => {
    setExpanded(!expanded);
  };

  const toggleEdit = () => {
    if (!expanded) {
      setExpanded(true);
    }
    setEditing(!editing);
  };

  const handleEditContent = ev => {
    ev.persist();
    console.log('this should be updated', ev.target.innerText);
    let newInputs = {
      ...inputs,
      [ev.target.attributes.name.value]: ev.target.innerText,
    };
    setInputs(newInputs);

    handleEditSubmit();
  };

  const handleEditSubmit = () => {
    let newBookmark = {
      ...inputs,
    };

    console.log(newBookmark);
  };
  const deleteBookmark = id => {
    ApiService.deleteBookmark(id)
      .then(removeBookmark(id))
      .catch(error => setError(error));
  };

  const renderError = () => {
    return (
      <>
        {error ? (
          <div className={s.error} aria-live="assertive">
            {error.message}
          </div>
        ) : null}
      </>
    );
  };

  const renderCollapsed = () => {
    let { title, rating, _id, url } = bookmark;
    return (
      <section className={s.bookmarkItem}>
        <div className={s.head}>
          <h3>{title}</h3>
        </div>
        <div className={s.rating}>
          <RatingSpan rating={rating} />
        </div>
        <div className={s.itemControlsContainer}>
          <div className={s.itemControls}>
            <div className={s.tooltip}>
              <MdEdit onClick={toggleEdit} />
              <span className={s.tooltiptext}>Edit</span>
            </div>
            <div className="tooltip delete">
              <MdDelete
                onClick={() => {
                  deleteBookmark(_id);
                }}
              />
              <span className={s.tooltiptext}>Delete</span>
            </div>
            <div className={s.tooltip}>
              <a
                rel="noopener noreferrer"
                target="_blank"
                href={url}
                className={s.linkButton}
              >
                <TiArrowForward />
              </a>
              <span className={s.tooltiptext}>Visit</span>
            </div>
          </div>
        </div>

        <div className={s.itemExpanded} onClick={() => handleExpand()}>
          <MdExpandMore />
          <span>More</span>
        </div>
      </section>
    );
  };

  const renderExpanded = () => {
    const { _id, title, desc, rating, url } = bookmark;

    return (
      <section id={_id} className={s.bookmarkItemExpanded}>
        <h3 className={s.headExpanded}>{title}</h3>
        <div className={s.ratingExpanded}>
          {editing ? (
            <select id={s.editRating}>
              <option value="">Rating</option>
              <option>1</option>
              <option>2</option>
              <option>3</option>
              <option>4</option>
              <option>5</option>
            </select>
          ) : (
            <RatingSpan rating={rating} />
          )}
        </div>

        <p
          className={s.url}
          role="textbox"
          suppressContentEditableWarning={true}
          contentEditable={editing}
        >
          {url}
        </p>

        <p
          className={s.desc}
          role="textbox"
          contentEditable={editing}
          suppressContentEditableWarning={true}
        >
          {desc ? desc : 'Add a description'}
        </p>

        <div className={s.itemControlsContainerExpanded}>
          <div className={s.itemControlsExpanded}>
            {editing ? (
              <div className={s.editControl}>
                <button>Cancel</button>
                <button onClick={handleEditSubmit}>Save</button>
              </div>
            ) : (
              <>
                <div className={s.less}>
                  <MdExpandLess onClick={() => handleExpand()} />
                  <span>Less</span>
                </div>
                <div className={s.tooltip}>
                  <MdEdit className={s.tooltip} onClick={() => toggleEdit()} />
                  <span className={s.tooltiptext}>Edit</span>
                </div>
                <div className={s.tooltip}>
                  <MdDelete
                    onClick={() => {
                      deleteBookmark(_id);
                    }}
                  />
                  <span className={s.tooltiptext}>Delete</span>
                </div>
                <div className={s.tooltip}>
                  <a
                    rel="noopener noreferrer"
                    target="_blank"
                    href={url}
                    className={s.linkButton}
                  >
                    <TiArrowForward />
                  </a>
                  <span className={s.tooltiptext}>Visit</span>
                </div>
              </>
            )}
          </div>
        </div>
      </section>
    );
  };

  const renderEditForm = () => {
    const { _id, title, desc, rating, url } = bookmark;

    return (
      <form className={s.editForm}>
        <fieldset>
          <div className={s.editTitle}>
            <label>Title</label>
            <input type="text" value={title} />
          </div>
          <div className={s.editRating}>
            <label>Rating</label>
            <select>
              <option value="">Rating</option>
              <option>1</option>
              <option>2</option>
              <option>3</option>
              <option>4</option>
              <option>5</option>
            </select>
          </div>
          <div className={s.editUrl}>
            <label>Url</label>
            <input type="text" value={url} />
          </div>

          <div className={s.editDesc}>
            <label>Description</label>
            <textarea className={s.desc} />
          </div>

          <div className={s.editControlsContainer}>
            <div className={s.editFormControls}>
              <button type="button">Cancel</button>
              <button type="submit">Save</button>
            </div>
          </div>
        </fieldset>
      </form>
    );
  };

  return (
    <>
      {editing
        ? renderEditForm()
        : expanded
        ? renderExpanded()
        : renderCollapsed()}
    </>
  );
}
