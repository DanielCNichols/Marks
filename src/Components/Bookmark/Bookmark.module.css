* {
  box-sizing: border-box;
  text-decoration: none;
  list-style: none;
}

.error {
  color: red;
  font-size: 0.8em;
}

.tooltiptext {
  display: none;
}

.bookmarkItem {
  width: 100%;
  display: grid;
  border: 2px solid white;
  grid-template-columns: 43% 43% 14%;
  grid-template-rows: 1fr 1fr 50px;
  background-color: #252525;
  height: auto;
  margin: 10px auto;
  padding: 0;
}

.head {
  grid-column-start: span 2;
  grid-row-start: 1;
  font-size: 1.3em;
  display: flex;
  justify-content: center;
  align-items: center;
  word-wrap: break-word;
}

.head h3 {
  margin: 10px 0;
  padding: 5px;
}

.rating {
  grid-column-start: span 2;
  grid-row-start: 2;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.1em;
  margin-bottom: 10px;
}

.itemControlsContainer,
r {
  display: flex;
  grid-column-start: 3;
  grid-column-end: 4;
  grid-row-start: span 3;
  margin-left: auto;
  width: 100%;
  background-color: #404140;
}

.itemControls {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  background-color: #404140;
  width: auto;
  margin: 0 auto;
}

.itemControls svg {
  margin: 10px 0;
  color: rgba(255, 255, 255, 0.8);
  font-size: 2em;
  padding: 5px;
}

.itemExpanded {
  grid-row-start: 3;
  grid-column-start: 1;
  grid-column-end: 3;
  text-align: left;
  width: 100%;
  padding-left: 5px;
  display: flex;
  align-items: center;
  border-top: 1px solid rgba(255, 255, 255, 0.3);
}

.itemExpanded svg {
  margin: 0 10px;
  font-size: 1.3em;
}

.bookmarkItemExpanded {
  width: 100%;
  display: grid;
  grid-template-columns: 1fr, 1fr;
  grid-template-rows: 1fr 1fr auto 50px;
  height: fit-content;
  border: 2px solid white;
  background-color: #252525;
  margin: 10px 0;
}

.headExpanded {
  grid-row-start: 1;
  word-wrap: break-word;
  text-align: left;
  display: flex;
  align-items: center;
  padding: 0 10px;
  border-bottom: 2px solid rgba(255, 255, 255, 0.5);
}

.headExpanded h3 {
  margin: 0;
}

.less {
  color: #23a840;
}

.ratingExpanded {
  grid-row-start: 1;
  grid-column-start: 2;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding: 0 20px;
  border-bottom: 2px solid rgba(255, 255, 255, 0.5);
}

.desc {
  grid-row-start: 3;
  grid-column-start: 1;
  grid-column-end: 3;
  padding: 0 10px;
  text-align: center;
  width: 100%;
  margin: 15px auto 25px auto;
}

.url {
  grid-row-start: 2;
  grid-column-start: span 2;
  padding: 0 10px;
  word-wrap: break-word;
  width: 100%;
  margin: 30px auto;
  text-align: center;
}

.itemControlsContainerExpanded {
  grid-row-start: 4;
  grid-column-start: span 2;
  border-top: 1px solid white;
  display: flex;
  align-items: center;
  background-color: #404140;
  width: 100%;
}

.itemControlsContainerExpanded svg {
  font-size: 1.3em;
}

.itemControlsExpanded {
  width: 100%;
  display: flex;
  color: rgba(255, 255, 255, 0.8);
  justify-content: space-around;
}

.less {
  display: flex;
  align-items: center;
}

@media screen and (min-width: 800px) {
  .bookmarkItem {
    width: 85%;
    max-width: 400px;
    min-width: 355px;
    grid-template-columns: 30% 70%;
    margin: 25px 0;
  }

  .bookmarkItemExpanded {
    width: 85%;
    max-width: 400px;
    min-width: 355px;
    margin: 25px 0;
  }

  .itemControlsContainer {
    display: initial;
    grid-column-start: 2;
    grid-row-start: 3;
    width: 100%;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    background-color: #404140;
  }

  .itemControls {
    flex-direction: row;
    align-items: center;
    justify-content: space-around;
    background-color: #404140;
    border-top: 1px solid rgba(255, 255, 255, 0.3);
    height: 100%;
  }

  .itemControls svg {
    padding: 0;
    margin: 0;
    font-size: 1.2em;
  }

  .tooltip {
    position: relative;
    width: auto;
    height: 50%;
  }

  .tooltiptext {
    visibility: hidden;
    width: auto;
    display: flex;
    background-color: black;
    color: rgba(255, 255, 255, 0.8);
    text-align: center;
    border-radius: 6px;
    padding: 5px 0;
    position: absolute;
    left: -70%;
    opacity: 0;
    transition: 0.5s;
    transition-delay: 0.5s;
  }

  .tooltip:hover .tooltiptext {
    visibility: visible;
    position: absolute;
    border: 1px solid rgba(255, 255, 255, 0.5);
    padding: 3px 5px;
    opacity: 1;
  }

  .itemControls div:nth-child(2) > svg:hover {
    color: red;
  }

  .itemControlsExpanded div:nth-child(3) > svg:hover {
    color: red;
  }

  .itemControls div:nth-child(3) > a:hover svg {
    color: #68cffa;
  }

  .itemControlsExpanded div:nth-child(4) > a:hover svg {
    color: #68cffa;
  }

  .itemControls svg:nth-child(1):hover {
    color: #23a841;
  }

  .itemControlsExpanded div:nth-child(2) > svg:hover {
    color: #23a841;
  }

  .itemExpanded {
    grid-column-start: 1;
    grid-column-end: 2;
    background-color: #404140;
  }

  .itemExpanded svg {
    margin: 0;
  }
}
