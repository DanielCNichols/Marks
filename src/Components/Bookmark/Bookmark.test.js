import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Bookmark from './Bookmark';

const bookmark = {
  title: 'test title',
  desc: 'A test bookmark',
  rating: 3,
  url: 'https://www.twitter.com',
};

describe('Bookmark tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Bookmark bookmark={bookmark} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer.create(<Bookmark bookmark={bookmark} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
