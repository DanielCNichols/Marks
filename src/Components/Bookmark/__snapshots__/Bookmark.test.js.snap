// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark tests renders the UI as expected 1`] = `
<section
  className="bookmarkItem"
>
  <div
    className="head"
  >
    <h3>
      test title
    </h3>
  </div>
  <div
    className="rating"
  >
    <div
      className="rating"
    >
      <span>
        <svg
          className="filled"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M463 192H315.9L271.2 58.6C269 52.1 262.9 48 256 48s-13 4.1-15.2 10.6L196.1 192H48c-8.8 0-16 7.2-16 16 0 .9.1 1.9.3 2.7.2 3.5 1.8 7.4 6.7 11.3l120.9 85.2-46.4 134.9c-2.3 6.5 0 13.8 5.5 18 2.9 2.1 5.6 3.9 9 3.9 3.3 0 7.2-1.7 10-3.6l118-84.1 118 84.1c2.8 2 6.7 3.6 10 3.6 3.4 0 6.1-1.7 8.9-3.9 5.6-4.2 7.8-11.4 5.5-18L352 307.2l119.9-86 2.9-2.5c2.6-2.8 5.2-6.6 5.2-10.7 0-8.8-8.2-16-17-16z"
          />
        </svg>
      </span>
      <span>
        <svg
          className="filled"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M463 192H315.9L271.2 58.6C269 52.1 262.9 48 256 48s-13 4.1-15.2 10.6L196.1 192H48c-8.8 0-16 7.2-16 16 0 .9.1 1.9.3 2.7.2 3.5 1.8 7.4 6.7 11.3l120.9 85.2-46.4 134.9c-2.3 6.5 0 13.8 5.5 18 2.9 2.1 5.6 3.9 9 3.9 3.3 0 7.2-1.7 10-3.6l118-84.1 118 84.1c2.8 2 6.7 3.6 10 3.6 3.4 0 6.1-1.7 8.9-3.9 5.6-4.2 7.8-11.4 5.5-18L352 307.2l119.9-86 2.9-2.5c2.6-2.8 5.2-6.6 5.2-10.7 0-8.8-8.2-16-17-16z"
          />
        </svg>
      </span>
      <span>
        <svg
          className="filled"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M463 192H315.9L271.2 58.6C269 52.1 262.9 48 256 48s-13 4.1-15.2 10.6L196.1 192H48c-8.8 0-16 7.2-16 16 0 .9.1 1.9.3 2.7.2 3.5 1.8 7.4 6.7 11.3l120.9 85.2-46.4 134.9c-2.3 6.5 0 13.8 5.5 18 2.9 2.1 5.6 3.9 9 3.9 3.3 0 7.2-1.7 10-3.6l118-84.1 118 84.1c2.8 2 6.7 3.6 10 3.6 3.4 0 6.1-1.7 8.9-3.9 5.6-4.2 7.8-11.4 5.5-18L352 307.2l119.9-86 2.9-2.5c2.6-2.8 5.2-6.6 5.2-10.7 0-8.8-8.2-16-17-16z"
          />
        </svg>
      </span>
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M463 192H315.9L271.2 58.6C269 52.1 262.9 48 256 48s-13 4.1-15.2 10.6L196.1 192H48c-8.8 0-16 7.2-16 16 0 .9.1 1.9.3 2.7.2 3.5 1.8 7.4 6.7 11.3l120.9 85.2-46.4 134.9c-2.3 6.5 0 13.8 5.5 18 2.9 2.1 5.6 3.9 9 3.9 3.3 0 7.2-1.7 10-3.6l118-84.1 118 84.1c2.8 2 6.7 3.6 10 3.6 3.4 0 6.1-1.7 8.9-3.9 5.6-4.2 7.8-11.4 5.5-18L352 307.2l119.9-86 2.9-2.5c2.6-2.8 5.2-6.6 5.2-10.7 0-8.8-8.2-16-17-16zm-127.2 92.5c-10 7.2-14.2 20.2-10.2 31.8l30.1 87.7c1.3 3.7-2.9 6.8-6.1 4.6l-77.4-55.2c-4.9-3.5-10.6-5.2-16.3-5.2-5.7 0-11.4 1.7-16.2 5.2l-77.4 55.1c-3.2 2.3-7.4-.9-6.1-4.6l30.1-87.7c4-11.8-.2-24.8-10.3-32l-81-57.1c-3.2-2.2-1.6-7.3 2.3-7.3H196c12 0 22.7-7.7 26.5-19.1l29.6-88.2c1.2-3.6 6.4-3.6 7.6 0l29.6 88.2c3.8 11.4 14.5 19.1 26.5 19.1h97.3c3.9 0 5.5 5 2.3 7.2l-79.6 57.5z"
          />
        </svg>
      </span>
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M463 192H315.9L271.2 58.6C269 52.1 262.9 48 256 48s-13 4.1-15.2 10.6L196.1 192H48c-8.8 0-16 7.2-16 16 0 .9.1 1.9.3 2.7.2 3.5 1.8 7.4 6.7 11.3l120.9 85.2-46.4 134.9c-2.3 6.5 0 13.8 5.5 18 2.9 2.1 5.6 3.9 9 3.9 3.3 0 7.2-1.7 10-3.6l118-84.1 118 84.1c2.8 2 6.7 3.6 10 3.6 3.4 0 6.1-1.7 8.9-3.9 5.6-4.2 7.8-11.4 5.5-18L352 307.2l119.9-86 2.9-2.5c2.6-2.8 5.2-6.6 5.2-10.7 0-8.8-8.2-16-17-16zm-127.2 92.5c-10 7.2-14.2 20.2-10.2 31.8l30.1 87.7c1.3 3.7-2.9 6.8-6.1 4.6l-77.4-55.2c-4.9-3.5-10.6-5.2-16.3-5.2-5.7 0-11.4 1.7-16.2 5.2l-77.4 55.1c-3.2 2.3-7.4-.9-6.1-4.6l30.1-87.7c4-11.8-.2-24.8-10.3-32l-81-57.1c-3.2-2.2-1.6-7.3 2.3-7.3H196c12 0 22.7-7.7 26.5-19.1l29.6-88.2c1.2-3.6 6.4-3.6 7.6 0l29.6 88.2c3.8 11.4 14.5 19.1 26.5 19.1h97.3c3.9 0 5.5 5 2.3 7.2l-79.6 57.5z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="itemControlsContainer"
  >
    <div
      className="itemControls"
    >
      <div
        className="tooltip"
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
        <span
          className="tooltiptext"
        >
          Edit
        </span>
      </div>
      <div
        className="tooltip delete"
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
        <span
          className="tooltiptext"
        >
          Delete
        </span>
      </div>
      <div
        className="tooltip"
      >
        <a
          className="linkButton"
          href="https://www.twitter.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            baseProfile="tiny"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            version="1.2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 5.499c-.256 0-.512.097-.707.292-.195.196-.293.452-.293.708v2.559c-4.5.498-8 4.309-8 8.941v1c2.245-3.423 5.25-3.92 8-3.989v2.489c0 .255.098.512.293.707s.451.293.707.293.512-.098.707-.293l6.293-6.207-6.293-6.208c-.195-.195-.451-.292-.707-.292z"
            />
          </svg>
        </a>
        <span
          className="tooltiptext"
        >
          Visit
        </span>
      </div>
    </div>
  </div>
  <div
    className="itemExpanded"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
      />
    </svg>
    <span>
      More
    </span>
  </div>
</section>
`;
